pkg/cloud/aliyunprovider.go

```diff
@@ -0,0 +1,828 @@
+	aliyun.DownloadPricingDataLock.RLock()
+	defer aliyun.DownloadPricingDataLock.RUnlock()
+
+	keyFeature := pvk.Features()
+
+	pricing, ok := aliyun.Pricing[keyFeature]
+
+	if !ok {
+		log.Infof("Persistent Volume pricing not found for PV with feature: %s", keyFeature)
+		return &PV{}, nil
+	}
+
+	log.Infof("returing the PV price for the node with feature: %s", keyFeature)
+	return pricing.PV, nil
+}
+
+// Stubbed NetworkPricing for Aliyun. Will look at this in Next PR
+func (aliyun *Aliyun) NetworkPricing() (*Network, error) {
+	return &Network{
+		ZoneNetworkEgressCost:     0.0,
+		RegionNetworkEgressCost:   0.0,
+		InternetNetworkEgressCost: 0.0,
+	}, nil
+}
+
+// Stubbed LoadBalancerPricing for Aliyun. Will look at this in Next PR
+func (aliyun *Aliyun) LoadBalancerPricing() (*LoadBalancer, error) {
+	return &LoadBalancer{
+		Cost: 0.0,
+	}, nil
+}
+
+func (aliyun *Aliyun) GetConfig() (*CustomPricing, error) {
+	c, err := aliyun.Config.GetCustomPricingData()
+	if err != nil {
+		return nil, err
+	}
+	if c.Discount == "" {
+		c.Discount = "0%"
+	}
+	if c.NegotiatedDiscount == "" {
+		c.NegotiatedDiscount = "0%"
+	}
+	if c.ShareTenancyCosts == "" {
+		c.ShareTenancyCosts = defaultShareTenancyCost
+	}
+
+	return c, nil
+}
+
+// Load once and cache the result (even on failure). This is an install time secret, so
+// we don't expect the secret to change. If it does, however, we can force reload using
+// the input parameter.
+func (aliyun *Aliyun) loadAliyunAuthSecretAndSetEnv(force bool) error {
+	if !force && aliyun.loadedAccessKey {
+		return nil
+	}
+
+	exists, err := fileutil.FileExists(authSecretPath)
+	if !exists || err != nil {
+		return fmt.Errorf("Failed to locate service account file: %s", authSecretPath)
+	}
+
+	result, err := ioutil.ReadFile(authSecretPath)
+	if err != nil {
+		return err
+	}
+
+	var ak *AliyunAccessKey
+	err = json.Unmarshal(result, &ak)
+	if err != nil {
+		return err
+	}
+
+	err = env.Set(env.AliyunAccessKeyIDEnvVar, ak.AccessKeyID)
+	if err != nil {
+		return err
+	}
+	err = env.Set(env.AliyunAccessKeySecretEnvVar, ak.SecretAccessKey)
+	if err != nil {
+		return err
+	}
+	aliyun.loadedAccessKey = true
+	aliyun.accessKey = &credentials.AccessKeyCredential{
+		AccessKeyId:     ak.AccessKeyID,
+		AccessKeySecret: ak.SecretAccessKey,
+	}
+	return nil
+}
+
+// Regions returns a current supported list of Alibaba regions
+func (aliyun *Aliyun) Regions() []string {
+	return aliyunRegions
+}
+
+// ClusterInfo returns information about ALiyun cluster, as provided by metadata. TO-DO: Look at this function closely at next PR iteration
+func (aliyun *Aliyun) ClusterInfo() (map[string]string, error) {
+
+	c, err := aliyun.GetConfig()
+	if err != nil {
+		log.Errorf("Error opening config: %s", err.Error())
+	}
+
+	var clusterName string
+	if c.ClusterName != "" {
+		clusterName = c.ClusterName
+	}
+
+	// Use a default name if none has been set until this point
+	if clusterName == "" {
+		clusterName = "Aliyun Cluster #1"
+	}
```

Comment:
Why do we have a default at all? Is it any worse to use empty string, or `env.GetClusterID()` for this? I know I've heard SEs talk about recommending cluster name and ID be set the same, but there may be some quirk of logic here that I'm not aware of.